packages/babel-plugin-transform-react-constant-elements/src/index.ts

```diff
@@ -212,7 +212,19 @@ export default declare((api, options: Options) => {
         HOISTED.set(path.node, targetScope);
 
         // Only hoist if it would give us an advantage.
-        if (targetScope === jsxScope) return;
+        for (let currentScope = jsxScope; ; ) {
```

Comment:
This is now more complex, because in cases like
```js
function f() {
  let Component;
  {
    return <Component />;
  }
}
```
`jsxScope` is the block statement while `targetScope` is the function: we still do not want to hoist the element, because it's unnecessary.